infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -447,6 +477,7 @@ private void fetchTopologyFromClusterManager() {
      */
     private void stopLogReplication() {
         if (localClusterDescriptor.getRole() == ClusterRole.ACTIVE && isLeader.get()) {
+            isLeader.set(false);
```

Comment:
stopLogReplication doesn't mean leader loss, right?

Reply:
We should ideally call it in processLockRelease but there is an if condition in this method: 
`localClusterDescriptor.getRole() == ClusterRole.ACTIVE && isLeader.get()`
that will not get executed if we do it there.

Reply:
It used to be just: `if (localClusterDescriptor.getRole() == ClusterRole.ACTIVE`. Not sure why it was added later.